### ISSUE TITLE
Flakey http tests `tyrian.HttpTests`

### ISSUE THREAD

**Issue by davesmith00000** [PurpleKingdomGames/tyrian#199]
They are pretty slow and fail intermittently. Here for example the HEAD test passed, but I've seen it fail too, despite no code changes in that area.

```
tyrian.HttpTests:
  + HEAD 1.33s
  + POST 1.09s
  + PUT 3.28s
  + PATCH 0.40s
  + DELETE 1.65s
==> X tyrian.HttpTests.GET with headers 0.93s munit.ComparisonFailException: 
```

**Comment by davesmith00000** [comment]
Annoyingly I can't replicate this now. Will leave the issue open a bit longer...

**Comment by JPonte** [comment]
Seems like the tests I added 😞 I wonder if httpbin was having issues at the time? That's the downside of depending on an external service for the unit tests...

**Comment by davesmith00000** [comment]
Yeah it's being flaky again this morning - no worries though, it's still a massive improvement to have tests for this stuff!

I think maybe I'll just have to switch httpbin out for some sort of mock when I get a chance.

**Comment by davesmith00000** [comment]
I've disable the tests for now and left the following comment in the code while I have a think about what to do next:

```
    // These tests work great, but http://httpbin.org is unreliable, it seems.
    // So we either need an HTTP mock server compatible with Scala.js, or to use
    // containers (I'd rather not, but...). In the meantime these failing tests
    // are lowering build confidence, so I've disabled them.
```

I do want to bring them back though! Just not quite sure how best to proceed. httpbin is available as a Docker image, but I really dislike docker. :yum:

**Comment by JPonte** [comment]
Understandable, I also thought about some sbt magic that would start the httpbin container just for the tests but I didn't like the idea and my sbt knowledge is also questionable. I think @armanbilge had some fancy setup in their project with some simple http4s server running only for the tests or something? Maybe we can make an http4sbin 😅

**Comment by davesmith00000** [comment]
> http4sbin

The thought crossed my mind... 😄

**Comment by davesmith00000** [comment]
I happened to be manually testing an http call to httpbin this morning and it was a bit slow. I wonder if we could get away with just having retries with a backoff policy...

**Comment by JPonte** [comment]
Were you thinking of adding the retry policy to the API itself or just for the tests?
